src/coreclr/crossgen-corelib.proj

```diff
@@ -3,21 +3,107 @@
   <Import Project="Directory.Build.targets" />
 
   <Target Name="Build">
-    <ItemGroup>
-      <_CoreClrBuildArg Condition="'$(TargetArchitecture)' != ''" Include="-$(TargetArchitecture)" />
-      <_CoreClrBuildArg Include="-$(Configuration.ToLower())" />
-      <_CoreClrBuildArg Condition="!$([MSBuild]::IsOsPlatform(Windows))" Include="-os $(TargetOS)" />
-      <_CoreClrBuildArg Condition="'$(CrossBuild)' == 'true'" Include="-cross" />
-      <_CoreClrBuildArg Condition="'$(PortableBuild)' != 'true'" Include="-portablebuild=false" />
-    </ItemGroup>
+    <PropertyGroup>
+      <!-- When the 'Crossgen1' or 'Crossgen2' environment variable it set to 'true', -->
+      <!-- it will force the script to use the specified compiler to build System.Private.CoreLib. -->
+      <!-- In absence of these variables, the script uses its own default. -->
+      <UseCrossgen2ByDefault>true</UseCrossgen2ByDefault>
+
+      <UseCrossgen2 Condition="'$(Crossgen1)' == 'true'">false</UseCrossgen2>
+      <UseCrossgen2 Condition="'$(Crossgen2)' == 'true'">true</UseCrossgen2>
+      <UseCrossgen2 Condition="'$(UseCrossgen2)' == ''">$(UseCrossgen2ByDefault)</UseCrossgen2>
+
+      <OSPlatformConfig>$(TargetOS).$(TargetArchitecture).$(Configuration)</OSPlatformConfig>
+      <RootBinDir>$(RepoRoot)\artifacts</RootBinDir>
+      <LogsDir>$(RootBinDir)\log</LogsDir>
+      <BinDir>$(RootBinDir)\bin\coreclr\$(OSPlatformConfig)</BinDir>
+      <IntermediatesDir>$(RootBinDir)\obj\coreclr\$(OSPlatformConfig)</IntermediatesDir>
+      <CrossGenCoreLibLog>$(LogsDir)\CrossgenCoreLib_$(TargetOS)__$(TargetArchitecture)__$(Configuration).log</CrossGenCoreLibLog>
+      <ExeExtension Condition="'$(OS)' == 'Windows_NT'">.exe</ExeExtension>
+      <DotNetCli>$(RepoRoot)/dotnet.sh</DotNetCli>
+      <DotNetCli Condition="'$(OS)' == 'Windows_NT'">$(RepoRoot)\dotnet.cmd</DotNetCli>
+
+      <CoreLibAssemblyName>System.Private.CoreLib</CoreLibAssemblyName>
+      <CoreLibInputPath>$(BinDir)\IL\$(CoreLibAssemblyName).dll</CoreLibInputPath>
+      <CoreLibOutputPath>$(BinDir)\$(CoreLibAssemblyName).dll</CoreLibOutputPath>
+
+      <CrossDir></CrossDir>
+      <CrossDir Condition="'$(TargetArchitecture)' == 'arm' or '$(TargetArchitecture)' == 'arm64'">x64</CrossDir>
+      <CrossDir Condition="'$(TargetArchitecture)' == 'arm' and !$(UseCrossgen2) and '$(OS)' == 'Windows_NT'">x86</CrossDir>
```

Comment:
Nit: `!$(UseCrossgen2)` may result in a syntax error if both `Crossgen1` and `Crossgen2` are not `'true'` and `UseCrossgen2` is set to something bogus.  I think for that reason we usually compare properties with `'true'` and `'false'` string literals instead of using boolean operators like `!`.

Reply:
Well, my initial expectation was that "UseCrossgen2" would be a "clean internal property" that's always guaranteed to be set to either "false" or "true". I believe the version of the script in my PR satisfies that. I can easily modify these clauses to text comparisons with 'true' if you believe it's easier to reason about though.

Reply:
You can make it clean by initializing it to the empty string: `<UseCrossgen2 />`.  As in in this PR, it may be overridden on the command line with `/p:UseCrossgen2=blah`.

Reply:
I do admit that in "my interpretation" UseCrossgen2 should be pre-set to an empty string first.